arangod/Cluster/ClusterInfo.cpp

```diff
@@ -2394,6 +2394,23 @@ Result ClusterInfo::checkCollectionPreconditions(std::string const& databaseName
   return {};
 }
 
+Result ClusterInfo::incrementPlanVersion() {
+  auto trx = AgencyWriteTransaction{
+    std::vector<AgencyOperation>{IncreaseVersion()}, std::vector<AgencyPrecondition>{}};
+  AgencyComm ac(_server);
+  auto res = ac.sendTransactionWithFailover(trx);
+  
+  LOG_TOPIC("ccb89", DEBUG, Logger::CLUSTER) << "incrementing plan version";
+
+  if (res.successful()) {
+    if (res.slice().get("results").length() > 0) {
+      Result r = waitForPlan(res.slice().get("results")[0].getNumber<uint64_t>()).get();
+      return r.fail() ? r : Result{};
+    }
+  }
+  return Result{TRI_ERROR_INTERNAL, "Failed to perform a simple plan version incrment."};
```

Comment:
I am not happy with returning TRI_ERROR_INTERNAL here. I guess this will be a rare error, however, when it is occur it is probably rather due to connectivity issues and not a programming error. Suggesting a different error code, but I don't know whether it is the optimal one:
```suggestion
  return Result{TRI_ERROR_CLUSTER_AGENCY_COMMUNICATION_FAILED, "Failed to perform a simple plan version increment."};
```